pkg/archive/archive.go

```diff
@@ -776,7 +776,7 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, o
 	}
 
 	var xattrErrs []string
-	for key, value := range hdr.Xattrs {
+	for key, value := range hdr.PAXRecords {
```

Comment:
Not all PAXRecords are xattrs. Applying all records as xattrs, especially without first stripping off the "SCHILY.xattr." prefix from the xattr name, is incorrect.